api/core/routes.py

```diff
@@ -302,42 +255,195 @@ def get_ephemeris_by_catalog_number_jdstep():
         abort(400) 
 
     #Cast the latitude, longitude, and jd to floats (request parses as a string)
-    lat = float(latitude)
-    lon = float(longitude)
-    ele = float(elevation)
+    try:
+        location = EarthLocation(lat=float(latitude)*u.deg, lon=float(longitude)*u.deg, height=float(elevation)*u.m)
+    except:
+        abort(500, "Error: Invalid location parameters")
     
     jd0 = float(startjd)
     jd1 = float(stopjd) 
     jds = float(stepjd)
   
-    jd = my_arange(jd0,jd1,jds)
+    jd = jd_arange(jd0,jd1,jds)
 
     if(len(jd)>1000):
         app.logger.info("Too many results requested")
         abort(400)
 
-    return create_result_list(lat, lon, ele, jd, catalog, True)
+    tleLine1, tleLine2, date_collected, name = get_TLE(catalog, True)
+    return create_result_list(location, jd, tleLine1, tleLine2, date_collected, name)
+
+@app.route('/ephemeris/tle/')
+@limiter.limit("100 per second, 2000 per minute", key_func=lambda:get_forwarded_address(request))
+def get_ephemeris_by_tle():
+    '''
+    Returns satellite location and velocity information relative to the observer's coordinates
+    for a given Two Line Element Data Set at the specified Julian Date.
+
+    **Please note, for the most accurate results, a Julian Date close to the TLE epoch is necessary.
+
+    Parameters
```

Comment:
I'm a little confused how the docs have several parameters nicely listed here, but the function itself takes none of them, and instead there is some flask magic happening later to verify all the parameters exist. Is there some way to pass the parameters via python for clarity (and possibly type checking) purposes?